bitcoin/src/pow.rs

```diff
@@ -1613,3 +1607,29 @@ mod tests {
         work_multiplication_by_u64, 2_u64;
     }
 }
+
+#[cfg(kani)]
+mod verification {
+    use super::*;
+
+    #[cfg(kani)]
+    #[kani::proof]
+    fn check_mul_128_by_u64_never_overflows() {
+        let x: u128 = kani::any();
+        let y: u64 = kani::any();
+
+        let x = U256::from(x);
+        let (_res, overflow) = x.mul_u64(y);
+        assert!(!overflow);
+    }
+
+    #[cfg(kani)]
+    #[kani::proof]
+    fn check_mul_u64() {
+        let high: u128 = kani::any();
+        let low: u128 = kani::any();
+        let y: u64 = kani::any();
+
+        let _ = U256(high, low).mul_u64(y);
```

Comment:
If there are more `kani` tests in the future we should impl `Arbitrary` for `U256`.